runtime/contract_update_validation.go

```diff
@@ -494,30 +494,33 @@ func (validator *ContractUpdateValidator) checkConformances(
 	newDecl *ast.CompositeDeclaration,
 ) {
 
+	// Here it is assumed enums will always have one and only one conformance.
+	// This is enforced by the checker.
+	// Therefore, below check for multiple conformances is only applicable
+	// for non-enum type composite declarations. i.e: structs, resources, etc.
+
 	oldConformances := oldDecl.Conformances
 	newConformances := newDecl.Conformances
 
-	if len(oldConformances) != len(newConformances) {
-		validator.report(&ConformanceCountMismatchError{
-			Expected: len(oldConformances),
-			Found:    len(newConformances),
-			Range:    ast.NewRangeFromPositioned(newDecl.Identifier),
-		})
-
-		// If the lengths are not the same, trying to match the conformance
-		// may result in too many regression errors. hence return.
-		return
-	}
+	// All the existing conformances must have a match. Order is not important.
+	// Having extra new conformance is OK. See: https://github.com/onflow/cadence/issues/1394
+	for _, oldConformance := range oldConformances {
+		found := false
+		for _, newConformance := range newConformances {
```

Comment:
Maybe we can optimize the quadratic search here to avoid blow-up (in a follow-up PR). I guess because the outer loop break on first missing, it is actually OK?

Reply:
Optimized it here: 3c22c34